python_modules/dagster/dagster_tests/definitions_tests/decorators_tests/test_asset_check_decorator.py

```diff
@@ -540,3 +560,62 @@ def check1(context: AssetExecutionContext):
         return AssetCheckResult(passed=True)
 
     execute_assets_and_checks(asset_checks=[check1], resources={"io_manager": DummyIOManager()})
+
+
+def test_partitioned_asset():
+    @asset(partitions_def=StaticPartitionsDefinition(["a", "b", "c"]))
+    def my_asset(context):
+        return context.partition_key
+
+    @asset_check(asset=my_asset)
+    def my_check(my_asset):
+        assert my_asset == {"a": "a", "b": "b", "c": "c"}
+        return AssetCheckResult(passed=True)
+
+    with instance_for_test() as instance:
+        execute_assets_and_checks(instance=instance, assets=[my_asset], partition_key="a")
+        execute_assets_and_checks(instance=instance, assets=[my_asset], partition_key="b")
+        execute_assets_and_checks(
+            instance=instance, assets=[my_asset], asset_checks=[my_check], partition_key="c"
+        )
+
+
+def test_partitioned_asset_db_io_manager(tmp_path):
```

Comment:
bit odd to be using duckdb in a definitions test, worth moving to a new suite?

Reply:
I agree that it's odd to use duckdb here. What exactly is the behavior that we're testing here?  Wondering if it would make sense for the test to use a dummy IO manager that has whatever the behavior is necessarily to exercise the functionality we're testing?  Or alternatively to include it in the dagster duckdb package library instead of we're trying to exercise behavior of that integration?

Reply:
Testing that io managers correctly handle the all to one partition mapping, including these db io managers which look at type hints. This isn't asset check specific but it doesn't feel bad to have some coverage with checks. I wanted to make sure for https://github.com/dagster-io/dagster/discussions/17194#discussioncomment-7567175. I'll stick it in the duckdb library, doesn't hurt